ksqldb-common/src/main/java/io/confluent/ksql/function/GenericsUtil.java

```diff
@@ -213,23 +219,32 @@ private static boolean resolveGenerics(
     if (schema instanceof LambdaType) {
       final LambdaType lambdaType = (LambdaType) schema;
       final SqlLambda sqlLambda = instance.getSqlLambdaOrThrow();
-      if (sqlLambda.getInputType().size() != lambdaType.inputTypes().size()) {
-        throw new KsqlException(
-            "Number of lambda arguments doesn't match between schema and sql type");
-      }
-
-      int i = 0;
-      for (final ParamType paramType : lambdaType.inputTypes()) {
-        if (!resolveGenerics(
-            mapping, paramType, SqlArgument.of(sqlLambda.getInputType().get(i))
-        )) {
-          return false;
+      if (lambdaType.inputTypes().size() == sqlLambda.getNumInputs()) {
```

Comment:
This piece of logic seems to be the same as in `ParamType#areCompatible` for lambdas, could we extract them into a shared util?

Reply:
Hmm. It is the same logic, both checks make recursive calls to their respective functions so for that reason, I think it still makes sense to keep them in separate loops. It looks like the base case here is to add a generic to the list and the base case in `paramTypes` is to check to see if the `schema` and `instance` are a primitive match

Reply:
It's very similar, but as Leah said, it's slightly different, I think having them separate is OK for now.